core-libs/setup/ssr/engine/cx-common-engine.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  CommonEngine,
+  CommonEngineOptions,
+  CommonEngineRenderOptions,
+} from '@angular/ssr';
+import { CxServerErrorResponse } from '../error-handling';
+import { PROPAGATE_SERVER_ERROR_RESPONSE } from '../error-handling/server-error-response/propagate-server-error-response';
+
+/**
+ * The Spartacus extension of the CommonEngine introduced to handle propagated server responses caught during server-side rendering.
+ * @extends {CommonEngine}
+ */
+export class CxCommonEngine extends CommonEngine {
+  /**
+   * Stores any server error response that occurs during the rendering process.
+   * @type {CxServerErrorResponse | undefined}
+   */
+  protected errorResponse: undefined | CxServerErrorResponse;
```

Comment:
CommonEngine's instance is one for all requests. We should not share his state among all requests. Otherwise it may lead to funky bugs.
Let's define it as a variable in the beginning of the `render()` method, so it's local and available only in the method's closure